### PR TITLE
DOCSUP-18631: Edit and translate to Russian (Encryption Codecs)

### PR DESCRIPTION
**Pull request by karnevil13**
Changelog category:

- Documentation #30448

### CODE REVIEW

File: docs/ru/sql-reference/statements/create/table.md [hunk 1]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
 ENGINE = MergeTree()
 ```
 
+### Кодеки шифрования {#create-query-encryption-codecs}
+
+Эти кодеки не сжимают данные, вместо этого они зашифровывают данные на диске. Воспользоваться кодеками можно, только когда ключ шифрования задан параметрами [шифрования] (../../ru/operations/server-configuration-parameters/settings.md#server-settings-encryption). Отметим, ставить кодеки шифрования имеет смысл в самый конец цепочки кодеков, потому что зашифрованные данные, как правило, нельзя сжать релевантным образом.
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:251]:
```suggestion
Эти кодеки не сжимают данные, вместо этого они зашифровывают данные на диске. Воспользоваться кодеками можно, только когда ключ шифрования задан параметрами [шифрования](../../ru/operations/server-configuration-parameters/settings.md#server-settings-encryption). Обратите внимание: ставить кодеки шифрования имеет смысл в самый конец цепочки кодеков, потому что зашифрованные данные, как правило, нельзя сжать релевантным образом.
```

File: docs/ru/sql-reference/statements/create/table.md [hunk 2]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
...
+
+Кодеки шифрования:
+
+-   `CODEC('AES-128-GCM-SIV')` — Зашифровывает данные с помощью AES-128 в режиме [RFC 8452](https://tools.ietf.org/html/rfc8452) GCM-SIV. 
+-   `CODEC('AES-256-GCM-SIV')` — Зашифровывает данные с помощью AES-256 в режиме GCM-SIV. 
+
+Эти кодеки используют фиксированный одноразовый код. Таким образом, это детерминированное шифрование. Оно совместимо с поддерживающими дедупликацию движками, в частности, [ReplicatedMergeTree](../../ru/engines/table-engines/mergetree-family/replication.md). Однако у шифрования имеется недостаток: если дважды зашифровать один и тот же блок данных, шифротекст на выходе получится одинаковым, и злоумышленник, у которого есть доступ к диску, заметит эту эквивалентность (при этом доступа к содержимому он не получит).
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:258]:
```suggestion
Эти кодеки используют фиксированный одноразовый ключ шифрования. Таким образом, это детерминированное шифрование. Оно совместимо с поддерживающими дедупликацию движками, в частности, [ReplicatedMergeTree](../../ru/engines/table-engines/mergetree-family/replication.md). Однако у шифрования имеется недостаток: если дважды зашифровать один и тот же блок данных, текст на выходе получится одинаковым, и злоумышленник, у которого есть доступ к диску, заметит эту эквивалентность (при этом доступа к содержимому он не получит).
```

File: docs/ru/sql-reference/statements/create/table.md [hunk 3]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
...
+
+!!! attention "Внимание"
+    Большинство движков, включая семейство "*MergeTree*", создают на диске индексные файлы, не применяя кодеки. А значит, в том случае, если зашифрованный столбец индексирован, на диске отобразится незашифрованный текст.
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:261]:
```suggestion
    Большинство движков, включая семейство `MergeTree`, создают на диске индексные файлы, не применяя кодеки. А значит, в том случае, если зашифрованный столбец индексирован, на диске отобразится незашифрованный текст.
```

File: docs/ru/sql-reference/statements/create/table.md [hunk 4]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
...
+
+!!! attention "Внимание"
+    Если вы выполняете запрос SELECT с упоминанием конкретного значения в зашифрованном столбце (например, при использовании секции WHERE), это значение может появиться в [system.query_log](../../ru/operations/system-tables/query_log.md). Рекомендуем отключить логирование.
+
+**Example**
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:266]:
```suggestion
**Пример**
```

File: docs/ru/sql-reference/statements/create/table.md [hunk 5]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
...
+```sql
+CREATE TABLE mytable 
+(
+    x String Codec(AES_128_GCM_SIV)
+) 
+ENGINE = MergeTree ORDER BY x;
+```
+
+!!!note "Замечание"
+    Если необходимо применить сжатие, это нужно эксплицитно прописать в запросе. Без этого произойдет лишь зашифровка данных.
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:277]:
```suggestion
    Если необходимо применить сжатие, это нужно явно прописать в запросе. Без этого будет выполнено только шифрование данных.
```

File: docs/ru/sql-reference/statements/create/table.md [hunk 6]
```diff
@@ -246,6 +246,46 @@ CREATE TABLE codec_example
...
+
+**Example**
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/create/table.md:279]:
```suggestion
**Пример**
```

**Review by FArthur-cmd — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Comment by olgarev** [comment]
@nikitamikhaylov Hi, would you merge it?

**Comment by nikitamikhaylov** [comment]
> @nikitamikhaylov Hi, would you merge it?

Yes, I will, but currently we have some internal troubles with out CI system and docs checks. For now they don't see any changes in docs even if they are here. From this PR: `INFO:root:No changes in documentation`. We have to fix that problem and then merge everything.

**Review by vitlibar — APPROVED** [review]

**Comment by FArthur-cmd** [comment]
@Mergifyio update

**Comment by FArthur-cmd** [comment]
@Mergifyio update

**Pull request merged**